File: Ejercicio 1/maximo_array.cpp
// maximo_array.cpp
// Este programa calcula el máximo de un array de números en C++

#include <iostream>
using namespace std;

// Función que recibe un array y su tamaño, y retorna el máximo
int maximoArray(int arr[], int n) {
    int max = arr[0];
    for (int i = 1; i < n; i++) {
        if (arr[i] > max) {
            max = arr[i];
        }
    }
    return max;
}

int main() {
    int numeros[] = {3, 7, 2, 9, 4};
    int n = sizeof(numeros) / sizeof(numeros[0]);
    cout << "El máximo es: " << maximoArray(numeros, n) << endl;
    return 0;
} 

File: Ejercicio 1/maximo_array.py
# maximo_array.py
# Este script calcula el máximo de un array de números en Python

# Definimos una función que recibe una lista de números y retorna el máximo

def maximo_array(arr):
    # Usamos la función incorporada max() para encontrar el máximo
    return max(arr)

# Ejemplo de uso
if __name__ == "__main__":
    numeros = [3, 7, 2, 9, 4]
    print("El máximo es:", maximo_array(numeros)) 